controllers/schedule_controller.go

```diff
@@ -99,14 +101,15 @@ func (r *BackupScheduleReconciler) Reconcile(
 		return ctrl.Result{}, err
 	}
 
+	// no velero schedules, so create them
 	if len(veleroScheduleList.Items) == 0 {
 		if err := r.initVeleroSchedules(ctx, backupSchedule, r.Client); err != nil {
 			return ctrl.Result{}, err
 		}
-	} else {
-		for _, veleroSchedule := range veleroScheduleList.Items {
-			updateScheduleStatus(ctx, &veleroSchedule, backupSchedule)
-		}
+	}
+	// update schedule status with latest velero schedules
+	for _, veleroSchedule := range veleroScheduleList.Items {
+		updateScheduleStatus(ctx, &veleroSchedule, backupSchedule)
 	}
 
 	err := r.Client.Status().Update(ctx, backupSchedule)
```

Comment:
Do we always need to update the status? Only if at line 112  the backupSchedule changed right? But we can add it later on...

Reply:
Sure, I'll compare to only update it if it's changed in my next PR :) Thanks!